src/k8s/cmd/k8s/k8s_x_snapd_config.go

```diff
@@ -41,17 +49,34 @@ func newXSnapdConfigCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 				return
 			}
 
-			switch mode.Orb {
-			case "none":
+			if mode.Orb == "none" {
 				cmd.PrintErrln("Warning: meta.orb is none, skipping reconcile actions")
 				return
-			case "k8sd":
-				client, err := env.Snap.K8sdClient("")
-				if err != nil {
-					cmd.PrintErrf("Error: failed to create k8sd client: %v\n", err)
+			}
+
+			client, err := env.Snap.K8sdClient("")
+			if err != nil {
+				cmd.PrintErrf("Error: failed to create k8sd client: %v\n", err)
+				env.Exit(1)
+				return
+			}
+
+			ctx, cancel := context.WithTimeout(cmd.Context(), opts.timeout)
+			defer cancel()
+			if err := control.WaitUntilReady(ctx, func() (bool, error) {
+				_, partOfCluster, err := client.NodeStatus(cmd.Context())
+				if !partOfCluster {
+					cmd.PrintErrf("Node is not part of a cluster: %v\n", err.Error())
```

Comment:
Is it possible that the query returns specifically this: https://github.com/canonical/k8s-snap/blob/f1c8937629405cd7e99f24215ad6e13e117092e6/src/k8s/pkg/client/k8sd/status.go#L19-L20
I know that this only occurs when the node has not been initialized, but if we ran into this, it would panic.

Reply:
Good catch @mateoflorido 
I removed the pointer deref and let `%v` just do its magic. It should automatically resolve to the error string if an error exists.